tests/virt_autotest/virt_utils.pm

```diff
@@ -379,4 +380,28 @@ sub is_installed_equal_upgrade_major_release {
     return $host_installed_version eq $host_upgrade_version;
 }
 
+#This subroutine is used for generating junit log, especially for those test modules outside virt_autotest
+#or use other bases instead of virt_autotest_base
+sub make_junit_log {
+    my ($self, $status, $runtime, $product_name, $package_name) = @_;
+    my $tc_result = virt_autotest_base::analyzeResult($self, $status, $runtime);
+    $self->{"product_tested_on"} = script_output("cat /etc/issue | grep -io \"SUSE.*\$(arch))\"");
+    $self->{"product_name"}      = $product_name;
+    $self->{"package_name"}      = $package_name;
+    @{$self->{success_guest_list}} = keys %xen::guests;
```

Comment:
This makes @{$self->{success_guest_list}} unreal. It should always reflect the real test results especially when more than one guests tested.

Reply:
This is experimental. I did not mean to do what you expected, so it can be changed.